### ISSUE TITLE
Using redacted values on errors/logs

### ISSUE THREAD

**Issue by brunomunizaf** [groue/GRDB.swift#1158]
Hello again @groue !

I was wondering if there is a flag I can set to hide the values of the requests which are present on errors that SQLite throws or even for trace logging. If there isn't, do you think this could be an interesting feature? Thanks again!

Best,
Bruno.

**Comment by groue** [comment]
Bonjour @brunomunizaf,

You can remove statement arguments in the trace:

```swift
db.trace { event in
    if case let .statement(statement) = event {
        print(statement.sql)
    }
}

// Prints SELECT * FROM player WHERE name = ?
_ = try Player.filter(Column("name") == "Alice").fetchOne(db)
_ = try Player.fetchOne(db, sql: "SELECT * FROM player WHERE name = ?", arguments: ["Alice"])
```

When you catch a [DatabaseError](https://github.com/groue/GRDB.swift/blob/master/README.md#databaseerror), you can log everything but the statement arguments (don't use the `description` property, and build your log from other properties).

Finally, the [Error Log](https://github.com/groue/GRDB.swift/blob/master/README.md#error-log) does not provide any way to redact its raw message.

**Comment by groue** [comment]
Thanks for again for asking this question. @brunomunizaf. Values should not be logged by default, and this will be fixed soon.

**Comment by groue** [comment]
See #1160

**Comment by groue** [comment]
#1160 has shipped in v5.20.0
